### PR TITLE
Login function

### PR DESCRIPTION
**Pull request by ikayz**
Login function to the heroku backend

### CODE REVIEW

File: src/components/Login.js [hunk 1]
```diff
@@ -1,89 +1,153 @@
...
 
-  handleEmailChange = (e) => {
-    this.setState({
-      email: e.target.value,
-    });
-  };
+				this.setState({
+					accessToken: res.token,
+				})
+				console.log(this.state.accessToken)
```
Comment (CIPHERTron) [inline src/components/Login.js:40]: this `console.log()` statement will log the access token @ikayz

File: src/components/Journal.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 import { Col, Row, Container, Table } from "react-bootstrap";
 
+import AddJournal from "./AddJournal";
+
 const Journal = () => {
   return (
     <Container>
       <h1>Journals</h1>
-      <Row>
+      <AddJournal />
```
Comment (yochannah) [inline src/components/Journal.js:10]: Should this be commented out?

File: src/components/Login.js [hunk 3]
```diff
@@ -1,35 +1,100 @@
...
+              <Form.Label>Email address</Form.Label>
+              <Form.Control
+                type="email"
+                placeholder="Enter email"
+                name="email"
+                value={this.state.email}
+                onChange={this.handleEmailChange}
+              />
+              <Form.Text className="text-muted">
+                We'll never share your email with anyone else.
```
Comment (yochannah) [inline src/components/Login.js:72]: Nice touch!

File: src/components/AddJournal.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+    const policies = {
+      title: policyTitle,
+      first_year: firstYear,
+      last_year: lastYear,
+      type: policyType,
+    };
+    const journal = { title, url, issn, rating, date, policies, domain };
+
+    fetch("https://journal-policy-tracker.herokuapp.com/api/journals", {
```
Comment (yochannah) [inline src/components/AddJournal.js:27]: Small request: This is fine for early dev, but now that we have multiple API calls in multiple places, let's try to move the URLs we're fetching from to a single config file. That way if we have to change hosts, we'll be able to switch quickly and only edit one single line.

File: src/components/Login.js [hunk 5]
```diff
@@ -1,35 +1,100 @@
...
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        email: this.state.email,
+        password: this.state.password,
+      }),
+    })
+      .then((response) => response.json())
+      .then((res) => {
+        console.log(res);
```
Comment (yochannah) [inline src/components/Login.js:27]: Let's remove this before we merge

File: src/components/Login.js [hunk 6]
```diff
@@ -1,35 +1,100 @@
...
+        // The response is expected to be in the form:
+        // {
+        // message: "Login Successful!"
+        // token: "",
+        // }
+
+        this.setState({
+          accessToken: res.token,
+        });
+        console.log(this.state.accessToken);
```
Comment (yochannah) [inline src/components/Login.js:37]: Is this log safe? (maybe yes -  please elaborate!)

**Review by yochannah — CHANGES_REQUESTED** [review]
A few comments and Qs - the major thing  I'd like  to see before merging is moving out the url into a single const or a config file.

File: src/components/AddJournal.js (hunk 4 above)
Comment (ikayz) [reply src/components/AddJournal.js:27]: Noted

File: src/components/Login.js (hunk 6 above)
Comment (ikayz) [reply src/components/Login.js:37]: That isn't supposed to be there

File: src/components/Journal.js (hunk 2 above)
Comment (ikayz) [reply src/components/Journal.js:10]: For now it's supposed to be there

**Comment by ikayz** [comment]
@yochannah the requested changes for these were fixed in the last commit. It should be good to go. I'm working on solving the other issues in readiness for deployment

**Pull request merged**